src/components/common/immatriculation/ImmatriculationHistoryEntry.vue

```diff
@@ -70,15 +75,16 @@
 
             async function downloadCertificate() {
                 isLoading.value = true;
-                await loadCertificate();
-                const fileLink = document.createElement("a");
-                fileLink.href = certificateDownloadURL.value;
-                fileLink.setAttribute("download", `${props.semester}.pdf`);
+                if (await loadCertificate()) {
```

Comment:
what if this fails?

Reply:
The call fails, if the responseHandler in the loadCertificate() function fails. In that case, a modal will be shown. I implemented the check here in order to prevent opening any download dialog for an empty file, if something failed when fetching the data.